src/main/java/stringcalculator/OperatorType.java

```diff
@@ -1,20 +1,59 @@
 package stringcalculator;
 
-import java.util.function.Function;
-
 public enum OperatorType {
-    PLUS_OPERATOR("+"),
-    MINUS_OPERATOR("-"),
-    MULTIPLY_OPERATOR("*"),
-    DIVIDE_OPERATOR("/");
+    PLUS_OPERATOR("+") {
+        @Override
+        public double calculate(double sum, double nextNumber) {
+            return sum + nextNumber;
+        }
+    },
+    MINUS_OPERATOR("-") {
+        @Override
+        public double calculate(double sum, double nextNumber) {
+            return sum - nextNumber;
+        }
+    },
+    MULTIPLY_OPERATOR("*") {
+        @Override
+        public double calculate(double sum, double nextNumber) {
+            return sum * nextNumber;
+        }
+    },
+    DIVIDE_OPERATOR("/") {
+        @Override
+        public double calculate(double sum, double nextNumber) {
+            if (nextNumber == 0) {
+                throw new IllegalArgumentException("0 으로 나눌수 없습니다.");
+            }
+            return sum / nextNumber;
+        }
+    };
 
     private String operator;
 
     OperatorType(String operator) {
         this.operator = operator;
     }
 
-    public boolean isEqual(String operator) {
-        return this.operator.equals(operator);
+    public static OperatorType createOperator(String operator) {
+        if(operator.equals(PLUS_OPERATOR.operator)) {
```

Comment:
IntelliJ IDEA의 코드 자동 정렬 기능(⌥⌘L, Ctrl+Alt+L)을 사용하면 더 깔끔한 코드를 볼 수 있을 거예요.

```suggestion
        if (operator.equals(PLUS_OPERATOR.operator)) {
```